### PR TITLE
[CHANGES REQUESTED !!!] EPMRPP-38241 || Project manager permissions update

### PR DESCRIPTION
**Pull request by Yumfriez**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java [hunk 1]
```diff
@@ -284,10 +286,12 @@ private void assignUser(String name, ProjectRole projectRole, List<String> assig
 		projectUser.setUser(modifyingUser);
 		projectUser.setProject(project);
 		project.getUsers().add(projectUser);
-		aclHandler.permitSharedObjects(project.getId(),
-				name,
-				ProjectRole.PROJECT_MANAGER.higherThan(projectRole) ? BasePermission.READ : BasePermission.ADMINISTRATION
-		);
+
+		List<Permission> permissions = Lists.newArrayList(BasePermission.READ);
+		if (!ProjectRole.PROJECT_MANAGER.higherThan(projectRole)) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java]: let's make this condition more understandable using for example 'lowerThan'

File: src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java [hunk 2]
```diff
@@ -82,9 +81,8 @@ public void initAcl(Object object, String owner, Long projectId, boolean isShare
 	public void updateAcl(Object object, Long projectId, boolean isShared) {
 		if (isShared) {
 			userRepository.findUsernamesWithProjectRolesByProjectId(projectId).forEach((key, value) -> {
-				if (ProjectRole.PROJECT_MANAGER.higherThan(value)) {
-					aclService.addPermissions(object, key, BasePermission.READ);
-				} else {
+				aclService.addPermissions(object, key, BasePermission.READ);
+				if (!ProjectRole.PROJECT_MANAGER.higherThan(value)) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java]: let's make this condition more understandable using for example 'lowerThan'

File: src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java [hunk 3]
```diff
@@ -61,9 +61,8 @@ public void initAcl(Object object, String owner, Long projectId, boolean isShare
 					.stream()
 					.filter(entry -> !entry.getKey().equalsIgnoreCase(owner))
 					.forEach(entry -> {
-						if (ProjectRole.PROJECT_MANAGER.higherThan(entry.getValue())) {
-							aclService.addPermissions(object, entry.getKey(), BasePermission.READ);
-						} else {
+						aclService.addPermissions(object, entry.getKey(), BasePermission.READ);
+						if (!ProjectRole.PROJECT_MANAGER.higherThan(entry.getValue())) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java]: let's make this condition more understandable using for example 'lowerThan'

File: src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java (hunk 1 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java]: SameOrLower method has been added

File: src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java (hunk 2 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java]: SameOrLower method has been added

File: src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java (hunk 3 above)
Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/auth/acl/ShareableObjectsHandler.java]: SameOrLower method has been added

File: src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java [hunk 4]
```diff
@@ -287,10 +289,12 @@ private void assignUser(String name, ProjectRole projectRole, List<String> assig
 		projectUser.setUser(modifyingUser);
 		projectUser.setProject(project);
 		project.getUsers().add(projectUser);
-		aclHandler.permitSharedObjects(project.getId(),
-				name,
-				ProjectRole.PROJECT_MANAGER.higherThan(projectRole) ? BasePermission.READ : BasePermission.ADMINISTRATION
-		);
+
+		List<Permission> permissions = Lists.newArrayList(BasePermission.READ);
+		if (ProjectRole.PROJECT_MANAGER.sameOrLowerThan(projectRole)) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java]: You've said that such sequence is used for preventing npe. But inside the method you call projectRole.value. So if you have null here you would have npe anyway. So probably it's better make null check here or somewhere upper and replace with 'projectRole.sameOrHigherThan(PROJECT_MANAGER)'

Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/core/project/impl/UpdateProjectHandlerImpl.java]: fixed

**Pull request merged**